modules/mini-games/Race.lua

```diff
@@ -260,25 +264,42 @@ local function Nth (n) return n..getSuffix(n) end
 
 --- Function called by mini game module to stop a race
 local function stop()
-    -- Print the place that each player came
     local results = {}
-    for name, value in pairs(scores["finish_times"]) do
-        local time = value[2]
-
-        local up_result = results[#results]
-        if up_result and up_result.score == math.round(time, 2) then
-            up_result.players[#up_result.players + 1] = name
-
-        else
-            results[#results + 1] = {
-                place = value[1],
-                score = math.round(time, 2),
-                players = {name}
-            }
+    local results_by_name = {}
+
+    -- Add scores for players that finished the game
+    for name, result in pairs(variables["finish_times"]) do
+        results[#results + 1] = {
+            place = result.place,
+            score = math.round(result.time, 2),
+            players = {name}
+        }
+        results_by_name[name] = results[#results]
+
+        -- Correct placement in case of score ties
+        if #results > 1 then
+            local prev = results[#results - 1]
+            local curr = results[#results]
+            if curr.score == prev.score then
+                curr.place = prev.place
+            end
+        end
+    end
+
+    -- Add additional score data for all participants
+    for name, score in pairs(scores) do
+        if not results_by_name[name] then
+            results[#results + 1] = { players = {name} }
+            results_by_name[name] = results[#results]
```

Comment:
Same as above comment, but also:
So is place no longer a required field for the results data?

Reply:
Yes, place and score is optional.  As noted in the OP its to pass extra result data for players/teams that did not finish in time to get a score or left the game early.

Reply:
Thanks for the clarification, does this mean we now have a time limit for the race implimented, or this is just preparation for this feature?

Reply:
There is no time limit yet.